### ISSUE TITLE
I have an issue with Arabic Language. Font doesn't work very well. 

### ISSUE THREAD

**Issue by Faisalcs** [fastlane/fastlane#7522]
This is how it looks like when I run frameit on Arabic language. Does frameit support all languages ? 
![iphone6-0launch_framed](https://cloud.githubusercontent.com/assets/1807889/21249108/d8bbc9b8-c34c-11e6-8ff8-f548e5b8c259.png)

**Comment by Faisalcs** [comment]
<details><summary>🚫 fastlane environment 🚫</summary>

### Stack

| Key                 | Value                                       |
| ------------------- | ------------------------------------------- |
| OS                  | 10.12.2                                     |
| Ruby                | 2.2.4                                       |
| Bundler?            | false                                       |
| Git                 | git version 2.6.2                           |
| Installation Source | ~/.fastlane/bin/bundle/bin/fastlane         |
| Host                | Mac OS X 10.12.2 (16C67)                    |
| Ruby Lib Dir        | ~/.fastlane/bin/bundle/lib                  |
| OpenSSL Version     | OpenSSL 1.0.2g  1 Mar 2016                  |
| Is contained        | true                                        |
| Xcode Path          | /Applications/Xcode.app/Contents/Developer/ |
| Xcode Version       | 8.2                                         |


### System Locale

| Variable | Value       |   |
| -------- | ----------- | - |
| LANG     | en_US.UTF-8 | ✅ |
| LC_ALL   | en_US.UTF-8 | ✅ |
| LANGUAGE | en_US.UTF-8 | ✅ |


### fastlane files:

<details><summary>`./fastlane/Fastfile`</summary>

```ruby
# Customise this file, documentation can be found here:
# https://github.com/fastlane/fastlane/tree/master/fastlane/docs
# All available actions: https://github.com/fastlane/fastlane/blob/master/fastlane/docs/Actions.md
# can also be listed using the `fastlane actions` command

# Change the syntax highlighting to Ruby
# All lines starting with a # are ignored when running `fastlane`

# If you want to automatically update fastlane if a new version is available:
# update_fastlane

# This is the minimum version number required.
# Update this, if you use features of a newer version
fastlane_version "1.111.0"

default_platform :ios

platform :ios do
  before_all do
    # ENV["SLACK_URL"] = "https://hooks.slack.com/services/..."
    cocoapods
    carthage
  end

  desc "Runs all the tests"
  lane :test do
    scan
  end

  lane :screenshots do
    snapshot
    frameit(white: true)
    deliver
  end


  desc "Submit a new Beta Build to Apple TestFlight"
  desc "This will also make sure the profile is up to date"
  lane :beta do
    # match(type: "appstore") # more information: https://codesigning.guide
    gym # Build your app - more options available
    pilot

    # sh "your_script.sh"
    # You can also use other beta testing services here (run `fastlane actions`)
  end

  desc "Deploy a new version to the App Store"
  lane :release do
    # match(type: "appstore")
    # snapshot
    gym # Build your app - more options available
    deliver(force: true)
    # frameit
  end



  # You can define as many lanes as you want

  after_all do |lane|
    # This block is called, only if the executed lane was successful

    # slack(
    #   message: "Successfully deployed new App Update."
    # )
  end

  error do |lane, exception|
    # slack(
    #   message: exception.message,
    #   success: false
    # )
  end
end


# More information about multiple platforms in fastlane: https://github.com/fastlane/fastlane/blob/master/fastlane/docs/Platforms.md
# All available actions: https://github.com/fastlane/fastlane/blob/master/fastlane/docs/Actions.md

# fastlane reports which actions are used
# No personal data is recorded. Learn more at https://github.com/fastlane/enhancer
```
</details>

<details><summary>`./fastlane/Appfile`</summary>

```ruby
app_identifier "io.faisal.fastlane" # The bundle identifier of your app
apple_id "faisal.it@icloud.com" # Your Apple email address

team_id "EVFUVDRURX"  # Developer Portal Team ID

# you can even provide different app identifiers, Apple IDs and team names per lane:
# More information: https://github.com/fastlane/fastlane/blob/master/fastlane/docs/Appfile.md
```
</details>

### fastlane gems

| Gem                 | Version | Update-Status       |
| ------------------- | ------- | ------------------- |
| cert                | 1.5.1   | ✅ Up-To-Date        |
| credentials_manager | 0.16.2  | ✅ Up-To-Date        |
| deliver             | 1.17.1  | ✅ Up-To-Date        |
| fastlane            | 2.0.2   | 🚫 Update availaible |
| fastlane_core       | 0.60.1  | ✅ Up-To-Date        |
| frameit             | 3.1.1   | ✅ Up-To-Date        |
| gym                 | 1.14.0  | ✅ Up-To-Date        |
| match               | 0.12.1  | ✅ Up-To-Date        |
| pem                 | 1.5.1   | ✅ Up-To-Date        |
| pilot               | 1.13.1  | ✅ Up-To-Date        |
| produce             | 1.5.1   | ✅ Up-To-Date        |
| scan                | 0.15.0  | ✅ Up-To-Date        |
| screengrab          | 0.6.1   | ✅ Up-To-Date        |
| sigh                | 1.13.1  | ✅ Up-To-Date        |
| snapshot            | 1.18.1  | ✅ Up-To-Date        |
| spaceship           | 0.39.0  | ✅ Up-To-Date        |
| supply              | 0.9.0   | ✅ Up-To-Date        |


### Loaded fastlane plugins:

**No plugins Loaded**


<details><summary><b>Loaded gems</b></summary>

| Gem                       | Version      |
| ------------------------- | ------------ |
| CFPropertyList            | 2.3.4        |
| activesupport             | 4.2.7.1      |
| addressable               | 2.5.0        |
| babosa                    | 1.0.2        |
| bigdecimal                | 1.2.6        |
| bundler                   | 1.13.6       |
| claide                    | 1.0.1        |
| cocoapods                 | 1.1.1        |
| cocoapods-core            | 1.1.1        |
| cocoapods-deintegrate     | 1.0.1        |
| cocoapods-downloader      | 1.1.2        |
| cocoapods-plugins         | 1.0.0        |
| cocoapods-search          | 1.0.0        |
| cocoapods-stats           | 1.0.0        |
| cocoapods-trunk           | 1.1.1        |
| cocoapods-try             | 1.1.0        |
| colored                   | 1.2          |
| commander                 | 4.4.1        |
| concurrent-ruby           | 1.0.2        |
| credentials_manager       | 0.16.2       |
| domain_name               | 0.5.20161129 |
| dotenv                    | 2.1.1        |
| escape                    | 0.0.4        |
| excon                     | 0.54.0       |
| faraday                   | 0.10.0       |
| faraday-cookie_jar        | 0.0.6        |
| faraday_middleware        | 0.10.1       |
| fastimage                 | 1.6.8        |
| fastlane_core             | 0.60.1       |
| fourflusher               | 2.0.1        |
| fuzzy_match               | 2.0.4        |
| gh_inspector              | 1.0.2        |
| google-api-client         | 0.9.20       |
| googleauth                | 0.5.1        |
| highline                  | 1.7.8        |
| http-cookie               | 1.0.3        |
| httpclient                | 2.8.3        |
| hurley                    | 0.2          |
| i18n                      | 0.7.0        |
| io-console                | 0.4.3        |
| json                      | 1.8.1        |
| jwt                       | 1.5.6        |
| krausefx-shenzhen         | 0.14.11      |
| libxml-ruby               | 2.9.0        |
| little-plugger            | 1.1.4        |
| logging                   | 2.1.0        |
| memoist                   | 0.15.0       |
| mime-types                | 3.1          |
| mime-types-data           | 3.2016.0521  |
| mini_magick               | 4.5.1        |
| minitest                  | 5.4.3        |
| molinillo                 | 0.5.4        |
| multi_json                | 1.12.1       |
| multi_xml                 | 0.6.0        |
| multipart-post            | 2.0.0        |
| nanaimo                   | 0.2.3        |
| nap                       | 1.1.0        |
| net-sftp                  | 2.1.2        |
| net-ssh                   | 3.2.0        |
| netrc                     | 0.7.8        |
| nokogiri                  | 1.5.6        |
| os                        | 0.9.6        |
| plist                     | 3.2.0        |
| power_assert              | 0.2.2        |
| psych                     | 2.0.8        |
| public_suffix             | 2.0.4        |
| rake                      | 10.4.2       |
| rdoc                      | 4.2.0        |
| representable             | 2.3.0        |
| retriable                 | 2.1.0        |
| rouge                     | 1.11.1       |
| rubygems-update           | 2.5.2        |
| rubyzip                   | 1.2.0        |
| rubyzip                   | 1.1.7        |
| security                  | 0.1.3        |
| signet                    | 0.7.3        |
| slack-notifier            | 1.5.1        |
| sqlite3                   | 1.3.12       |
| terminal-notifier         | 1.7.1        |
| terminal-table            | 1.4.5        |
| test-unit                 | 3.0.8        |
| thread_safe               | 0.3.5        |
| tzinfo                    | 1.2.2        |
| uber                      | 0.0.15       |
| unf                       | 0.1.4        |
| unf_ext                   | 0.0.7.2      |
| unicode-display_width     | 1.1.2        |
| word_wrap                 | 1.0.0        |
| xcode-install             | 2.0.9        |
| xcodeproj                 | 1.4.1        |
| xcpretty                  | 0.2.4        |
| xcpretty-travis-formatter | 0.0.4        |
</details>


*generated on:* **2016-12-16**
</details>

**Comment by Faisalcs** [comment]
Hi everyone, Is there anyone can help me please? I change the font but still doesn't work?

**Comment by hjanuschka** [comment]
@Faisalcs can you share some sample - files so we can reproduce on local machines?
also it would be awesome to mark the parts in the screenshot which are wrong, and maybe paint overlays on how it should look (as it is hard to understand, without any knowlegde about arabic characters)

**Comment by Faisalcs** [comment]
@hjanuschka Sorry for late replay. I uploaded the picture again with highlighting Arabic language part. 
![d8bbc9b8-c34c-11e6-8ff8-f548e5b8c259](https://cloud.githubusercontent.com/assets/1807889/21466891/82f17d2c-c9e9-11e6-9e79-172a95558e4c.png)

**Comment by hjanuschka** [comment]
you mean that it is white instead of black font color?
tbh. i still have troubles to fully understand the problem, because i have no clue of the characters.

can you share a folder with screenshots and a Framefile, so that i can reproduce on my machine?

**Comment by Faisalcs** [comment]
@hjanuschka No. Frameit separate characters from each other. It's should be one word combined not each characters detached from the word.

**Comment by hjanuschka** [comment]
@Faisalcs ok i think now i am on the right track, would you be so kind to copy &  paste the arabic string in here - so i can copy it and run it on my machine? (or could u share the `frameit` command).


i looked at the code, and the only plausible thing that comes to my mind is a ltr problem inside minimagick, but need to test this locally 👍

**Comment by Faisalcs** [comment]
@hjanuschka This is Arabic String : أهلا وسهلا … شكرا لتعاونك معنا  
and this is my Framefile 

{
  "default": {
    "keyword": {
      "fonts": [
        {
          "font": "./fonts/Arial.ttf",
          "supported": ["ar"]
        },
        {
          "font": "./fonts/Arial.ttf",
          "supported": ["en-US"]
        }
      ],
      "color": "#545454"
    },
    "title": {
      "fonts": [
        {
          "font": "./fonts/Arial.ttf",
          "supported": ["ar"]
        },
        {
          "font": "./fonts/Arial.ttf",
          "supported": ["en-US"]
        }
      ],
      "color": "#545454"
    },
    "background": "./background.jpg",
    "padding": 50
  },


  "data": [
    {
      "filter": "Brainstorming",
      "keyword": {
        "color": "#d21559"
      }
    },
    {
      "filter": "Organizing",
      "keyword": {
        "color": "#feb909"
      }
    },
    {
      "filter": "Sharing",
      "keyword": {
        "color": "#aa4dbc"
      }
    },
    {
      "filter": "Styling",
      "keyword": {
        "color": "#31bb48"
      }
    }
  ]
}

I think the issue comes from minimagick too. 
the command I use to run frameit tool : fastlane frameit

**Comment by hjanuschka** [comment]
ok i really hard tried this, and well i am finaly able to reproduce.
but i still have no fix for it, do you have any arabic ttf font file? as most of the guys on the imagemagick forums refere to use a true arabic font, i have not yet found one, could you name one?

**Comment by Faisalcs** [comment]
@hjanuschka You can use Arial.ttf. It's on the OS. Go to Font -> search for Arial.ttf. 

I think the issue comes from imagemagick tool too. I hope not.

**Comment by hjanuschka** [comment]
ok i used arial on my testings and yes it comes down to imagemagick - and i well i have not yet found any fix for imagemagick - there are a ton of resources around the internet about users having problems with RTL languages, and all the proposed solutions didn't work for me (like reversing the string before passing it to imagemagick)

do you have any imagemagick stuff in any programming language with a working arabic font output, i can look at?

tbh, i am really sorry for you, but i might not be able to help you in this issue.

**Comment by Faisalcs** [comment]
@hjanuschka unfortunately, I don't 😔. 

Thank you so much for your effort and help. I'll do my best to fix the issue.

**Comment by hjanuschka** [comment]
@Faisalcs please let me know if you find a solution.

**Comment by iMokhles** [comment]
تم حل مشكلتك في هذه ال
pull request
Fixed: #8131
